src/utils/authorizations.ts

```diff
@@ -0,0 +1,142 @@
+  AuthorizedActions.BLANK_CERTIFY_STATE,
+  AuthorizedActions.BLANK_COMPLETING_PARTY,
+  AuthorizedActions.CONTINUATION_IN_FILING,
+  AuthorizedActions.CONTINUATION_OUT_FILING,
+  AuthorizedActions.COOP_CORRECTION_FILING,
+  AuthorizedActions.CORP_CORRECTION_FILING,
+  AuthorizedActions.COURT_ORDER_POA,
+  AuthorizedActions.EDITABLE_CERTIFY_NAME,
+  AuthorizedActions.EDITABLE_COMPLETING_PARTY,
+  AuthorizedActions.FILE_AND_PAY,
+  AuthorizedActions.FIRM_ADD_BUSINESS,
+  AuthorizedActions.FIRM_CONVERSION_FILING,
+  AuthorizedActions.FIRM_DISSOLUTION_FILING,
+  AuthorizedActions.FIRM_EDITABLE_COMPLETING_PARTY,
+  AuthorizedActions.FIRM_EDITABLE_EMAIL_ADDRESS,
+  AuthorizedActions.FIRM_NO_MIN_START_DATE,
+  AuthorizedActions.INCORPORATION_APPLICATION_FILING,
+  AuthorizedActions.NO_COMPLETING_PARTY_MESSAGE_BOX,
+  AuthorizedActions.NO_CONTACT_INFO,
+  AuthorizedActions.NOTICE_WITHDRAWAL_FILING,
+  AuthorizedActions.OTHER_DISSOLUTION_FILING,
+  AuthorizedActions.REGISTRATION_FILING,
+  AuthorizedActions.RESTORATION_FILING,
+  AuthorizedActions.SAVE_DRAFT,
+  AuthorizedActions.STAFF_BREADCRUMBS,
+  AuthorizedActions.STAFF_COMMENTS,
+  AuthorizedActions.STAFF_PAYMENT,
+  AuthorizedActions.THIRD_PARTY_CERTIFY_STMT
+]
+
+/**
+ * The roles if the user is Maximus Staff.
+ * Ultimately we won't need this list and we'll just check auth roles for everything.
+ */
+const MaximusStaffRoles = []
+
+/**
+ * The roles if the user is Contact Centre Staff.
+ * Ultimately we won't need this list and we'll just check auth roles for everything.
+ */
+const ContactCentreStaffRoles = []
+
+/**
+ * The roles if the user is SBC Field Office Staff (aka SBC Staff Tier 2).
+ * Ultimately we won't need this list and we'll just check auth roles for everything.
+ */
+const SbcFieldOfficeStaffRoles = [
+  AuthorizedActions.AMALGAMATION_FILING,
+  AuthorizedActions.BLANK_CERTIFY_STATE,
+  AuthorizedActions.BLANK_COMPLETING_PARTY,
+  AuthorizedActions.CONTINUATION_IN_FILING,
+  AuthorizedActions.EDITABLE_CERTIFY_NAME,
+  AuthorizedActions.FILE_AND_PAY,
+  AuthorizedActions.FIRM_DISSOLUTION_FILING,
+  AuthorizedActions.FIRM_EDITABLE_COMPLETING_PARTY,
+  AuthorizedActions.FIRM_EDITABLE_EMAIL_ADDRESS,
+  AuthorizedActions.INCORPORATION_APPLICATION_FILING,
+  AuthorizedActions.OTHER_DISSOLUTION_FILING,
+  AuthorizedActions.REGISTRATION_FILING,
+  AuthorizedActions.SAVE_DRAFT,
+  AuthorizedActions.SBC_BREADCRUMBS,
+  AuthorizedActions.THIRD_PARTY_CERTIFY_STMT
+]
+
+/**
+ * The roles if the user is none of the other types.
+ * Ultimately we won't need this list and we'll just check auth roles for everything.
+ */
+const DefaultRoles = [
+  AuthorizedActions.AMALGAMATION_FILING,
+  AuthorizedActions.CONTINUATION_IN_FILING,
+  AuthorizedActions.FILE_AND_PAY,
+  AuthorizedActions.FIRM_DISSOLUTION_FILING,
+  AuthorizedActions.INCORPORATION_APPLICATION_FILING,
+  AuthorizedActions.OTHER_DISSOLUTION_FILING,
+  AuthorizedActions.REGISTRATION_FILING,
+  AuthorizedActions.SAVE_DRAFT
+]
```

Comment:
Should we add roles and checks for all filings?
- AGM Extension
- AGM Location Change
- Amalgamation Out
- etc

See [spreadsheet]( https://docs.google.com/spreadsheets/d/13P4ZA19lJfXJX95N5OfQMu6jYuotDH6h2hDf2DnAv0E/edit?gid=1874127980#gid=1874127980). Note that the roles (authorized actions) are still under discussion -- we will try to combine some of them, for example, staff-only filings. Ref: [28529](https://app.zenhub.com/workspaces/entities---olga-65af15f59e89f5043c2911f7/issues/gh/bcgov/entity/28529).